src/main/java/org/prebid/server/bidder/huaweiads/HuaweiUtilRequest.java

```diff
@@ -0,0 +1,474 @@
+                .findFirst()
+                .orElse(null);
+
+        if (countryPresentInMap != null) {
+            return mapCountryCodeAlpha3ToAlpha2.get(countryPresentInMap);
+        }
+        if (country.length() > 2) {
+            return country.substring(0, 2);
+        }
+        return DEFAULT_COUNTRY_NAME;
+    }
+
+    private static String resolvePkgName(App appFromBidReq) {
+        String bundle = appFromBidReq.getBundle();
+        if (StringUtils.isBlank(bundle)) {
+            throw new PreBidException("resolvePkgName: app.bundle is empty");
+        }
+        return bundle;
+    }
+
+    private static String resolveLang(App appFromBidReq) {
+        String lang = appFromBidReq.getContent() != null ? appFromBidReq.getContent().getLanguage() : null;
+        if (StringUtils.isBlank(lang)) {
+            return "en";
+        }
+        return lang;
+    }
+
+    private static HuaweiNetwork resolveHuaweiNetwork(BidRequest bidRequest) {
+        Device device = bidRequest.getDevice();
+        if (device != null) {
+            return HuaweiNetwork.builder()
+                    .type(device.getConnectiontype() != null
+                            ? device.getConnectiontype()
+                            : DEFAULT_UNKNOWN_NETWORK_TYPE)
+                    .carrier(resolveCarrier(device))
+                    .cellInfoList(resolveCellInfoList(device))
+                    .build();
+        }
+        return null;
+    }
+
+    private static Integer resolveCarrier(Device device) {
+        String[] arrMccmnc = device.getMccmnc().split("-");
+        String str = arrMccmnc[0] + arrMccmnc[1];
+        if (str.equals("46000") || str.equals("46002") || str.equals("46007")) {
+            return 2;
+        } else if (str.equals("46001") || str.equals("46006")) {
+            return 1;
+        } else if (str.equals("46003") || str.equals("46005") || str.equals("46011")) {
+            return 3;
+        } else {
+            return 99;
+        }
+    }
+
+    private static List<HuaweiCellInfo> resolveCellInfoList(Device device) {
+        String[] arrMccmnc = device.getMccmnc().split("-");
+        if (arrMccmnc.length > 2) {
+            return List.of(HuaweiCellInfo.of(arrMccmnc[0], arrMccmnc[1]));
+        }
+        return null;
+    }
+
+    private static Regs resolveRegs(BidRequest bidRequest) {
+        Regs regs = bidRequest.getRegs();
+        if (regs != null && regs.getCoppa() >= 0) {
+            return Regs.of(regs.getCoppa(), null);
+        }
+        return null;
+    }
+
+    private static Geo resolveGeo(BidRequest bidRequest) {
+        Device device = bidRequest.getDevice();
+        Geo geo = device != null ? device.getGeo() : null;
+        if (geo != null) {
+            return Geo.builder()
+                    .lon(geo.getLon())
+                    .lat(geo.getLat())
+                    .accuracy(geo.getAccuracy())
+                    .build();
+        }
+        return null;
```

Comment:
invert condition to move this bunch of code out of if statement and make code clearer